rskj-core/src/main/java/org/ethereum/datasource/RocksDbDataSource.java

```diff
@@ -0,0 +1,364 @@
+        } finally {
+            profiler.stop(metric);
+            resetDbLock.writeLock().unlock();
+        }
+    }
+
+    private void openDb(Options options, Path dbPath) {
+        try {
+            db = RocksDB.open(options, dbPath.toString());
+
+            alive = true;
+        } catch (RocksDBException ioe) {
+            logger.error(ioe.getMessage(), ioe);
+            panicProcessor.panic("rocksdb", ioe.getMessage());
+            throw new RuntimeException("Can't initialize database");
+        }
+    }
+
+    public static Path getPathForName(String name, String databaseDir) {
+        if (Paths.get(databaseDir).isAbsolute()) {
+            return Paths.get(databaseDir, name);
+        } else {
+            return Paths.get(getProperty("user.dir"), databaseDir, name);
+        }
+    }
+
+    @Override
+    public boolean isAlive() {
+        try {
+            resetDbLock.readLock().lock();
+            return alive;
+        } finally {
+            resetDbLock.readLock().unlock();
+        }
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public byte[] get(byte[] key) {
+        Objects.requireNonNull(key);
+        Metric metric = profiler.start(Profiler.PROFILING_TYPE.DB_READ);
+        resetDbLock.readLock().lock();
+        try {
+            if (logger.isTraceEnabled()) {
+                logger.trace("~> RocksDbDataSource.get(): {}, key: {}", name, ByteUtil.toHexString(key));
+            }
+
+            try {
+                byte[] ret = db.get(key);
+                if (logger.isTraceEnabled()) {
+                    logger.trace("<~ RocksDbDataSource.get(): {}, key: {}, return length: {}", name, ByteUtil.toHexString(key), (ret == null ? "null" : ret.length));
+                }
+
+                return ret;
+            } catch (RocksDBException e) {
+                logger.error("Exception. Retrying again...", e);
+                try {
+                    byte[] ret = db.get(key);
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("<~ RocksDbDataSource.get(): {}, key: {}, return length: {}", name, ByteUtil.toHexString(key), (ret == null ? "null" : ret.length));
+                    }
+
+                    return ret;
+                } catch (RocksDBException e2) {
+                    logger.error("Exception. Not retrying.", e2);
+                    panicProcessor.panic("rocksdb", String.format("Exception. Not retrying. %s", e2.getMessage()));
+                    throw new RuntimeException("Couldn't get the data back for the given key");
+                }
+            }
+        } finally {
+            resetDbLock.readLock().unlock();
+            profiler.stop(metric);
+        }
+    }
+
+    @Override
+    public byte[] put(byte[] key, byte[] value) {
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(value);
+
+        Metric metric = profiler.start(Profiler.PROFILING_TYPE.DB_WRITE);
```

Comment:
new constant name for metrics

Reply:
What do you mean by `new constant name`? Could you give me an example?